ffmpeg-docker/Dockerfile

```diff
@@ -0,0 +1,130 @@
+
+RUN     apt-get -yqq update && \
+        apt-get install -yq --no-install-recommends ca-certificates expat libgomp1 pulseaudio && \
+        apt-get autoremove -y && \
+        apt-get clean -y
+
+FROM base as build
+
+ARG        PKG_CONFIG_PATH=/opt/ffmpeg/lib/pkgconfig
+ARG        LD_LIBRARY_PATH=/opt/ffmpeg/lib
+ARG        PREFIX=/opt/ffmpeg
+ARG        MAKEFLAGS="-j2"
+
+ENV         FFMPEG_VERSION=4.1     \
+            X264_VERSION=20170226-2245-stable \
+            SRC=/usr/local
+
+RUN      buildDeps="autoconf \
+                    automake \
+                    cmake \
+                    curl \
+                    bzip2 \
+                    libexpat1-dev \
+                    g++ \
+                    gcc \
+                    git \
+                    gperf \
+                    libtool \
+                    make \
+                    nasm \
+                    perl \
+                    pkg-config \
+                    python \
+                    libssl-dev \
+                    yasm \
+                    libpulse-dev \
+                    libxcb1-dev libxcb-shm0-dev   libxcb-xfixes0-dev \
+                    zlib1g-dev" && \
+        apt-get -yqq update && \
+        apt-get install -yq --no-install-recommends ${buildDeps}
+## x264 http://www.videolan.org/developers/x264.html
+RUN \
+        DIR=/tmp/x264 && \
+        mkdir -p ${DIR} && \
+        cd ${DIR} && \
+        curl -sL https://download.videolan.org/pub/videolan/x264/snapshots/x264-snapshot-${X264_VERSION}.tar.bz2 | \
+        tar -jx --strip-components=1 && \
+        ./configure --prefix="${PREFIX}" --enable-shared --enable-pic --disable-cli && \
+        make && \
+        make install && \
+        rm -rf ${DIR}
+
+## ffmpeg https://ffmpeg.org/
+RUN  \
+        DIR=/tmp/ffmpeg && mkdir -p ${DIR} && cd ${DIR} && \
+        curl -sLO https://github.com/etherlabsio/FFmpeg/archive/n${FFMPEG_VERSION}.tar.gz && \
+        tar -zx --strip-components=1 -f n${FFMPEG_VERSION}.tar.gz 
+
+
+RUN \
+        DIR=/tmp/ffmpeg && mkdir -p ${DIR} && cd ${DIR} && \
+        ./configure \
+        --disable-debug \
+        --disable-doc \
+        --disable-ffplay \
+        --disable-avresample \
+        --enable-shared \
+        --disable-libopencore-amrnb \
+        --disable-libopencore-amrwb \
+        --enable-gpl \
+        --enable-libxcb \
+        --disable-libass \
+        --disable-libfreetype \
+        --disable-libvidstab \
+        --disable-libmp3lame \
+        --disable-libopenjpeg \
+        --disable-libopus \
+        --disable-libtheora \
+        --disable-libvorbis \
+        --disable-libvpx \
+        --disable-libx265 \
+        --disable-libxvid \
+        --enable-libx264 \
+        --disable-nonfree \
+        --disable-openssl \
+        --disable-libfdk_aac \
+        --disable-libkvazaar \
+        --disable-libaom --extra-libs=-lpthread \
+        --enable-postproc \
+        --enable-small \
+        --enable-version3 \
```

Comment:
@lijo-jose do we need to do this ?

Reply:
--enable-version3 is not needed. Will try without this.